util/yang_test.go

```diff
@@ -1024,6 +1032,67 @@ func TestEnumeratedUnionTypes(t *testing.T) {
 		},
 		{
 			desc: "multi-level with mixed types",
+			types: []*yang.YangType{{
+				Name:    "string",
+				Pattern: []string{"^a.*$"},
+				Kind:    yang.Ystring,
+				Length: yang.YangRange{{
+					Min: yang.FromInt(10),
+					Max: yang.FromInt(20),
+				},
+				},
+			}, {
+				Name: "int",
+				Kind: yang.Yint32,
+			}, {
+				Name: "identityref",
+				Kind: yang.Yidentityref,
+			}, {
+				Name: "union",
+				Kind: yang.Yunion,
+				Type: []*yang.YangType{{
+					Name:    "string",
+					Pattern: []string{"^a.*$"},
+					Kind:    yang.Ystring,
+					Length: yang.YangRange{{
+						Min: yang.FromInt(10),
+						Max: yang.FromInt(20),
+					},
+					},
+				}, {
+					Name: "inner-int",
+					Kind: yang.Yint32,
+				}, {
+					Name: "inner-iref",
+					Kind: yang.Yidentityref,
+				}, {
+					Name: "enumeration",
+					Kind: yang.Yenum,
+				}},
+			}},
+			want: []*yang.YangType{{
+				Name: "identityref",
+				Kind: yang.Yidentityref,
+			}, {
+				Name: "inner-iref",
+				Kind: yang.Yidentityref,
+			}, {
+				Name: "enumeration",
+				Kind: yang.Yenum,
+			}},
+			wantParentTypes: []*yang.YangType{{
+				Name: "parent-type",
+				Kind: yang.Yunion,
+			}, {
+				Name: "inner-iref",
+				Kind: yang.Yidentityref,
+			}, {
```

Comment:
Can you explain why we end up with two different `parent-type` here? I presume it is because `parent-type` includes both an enumeration itself and has a child union that includes an enumeration too?

Reply:
I added a docstring comment for the test table's `wantSubsumingTypes` field. Let me know if this is clear enough.